main/ejml-ddense/src/org/ejml/dense/row/CommonOps_DDRM.java

```diff
@@ -2770,6 +2770,46 @@ public static DMatrixRMaj permuteRowInv( int[] pinv, DMatrixRMaj input, DMatrixR
         return output;
     }
 
+    /**
+     * Fill all rows in the specified range with zeros
+     *
+     * @param start Index of the first row (inclusivce) to be filled
+     * @param end Index of the last row (exclusive) to be filled
+     */
+    public static void zeroRows(DMatrixRMaj a, int start, int end) {
+        Arrays.fill(a.data, start * a.numCols, end * a.numCols, 0);
+    }
+
+    /**
+     * Shifts all rows up by the specified number of places
+     *
+     * @param a A matrix. Modified.
+     * @param c The rows are shifted by this number.
+     */
+    public static void shiftRowsUp(DMatrixRMaj a, int c) {
+        if (c < a.numRows) {
+            System.arraycopy(a.data, c * a.numCols, a.data, 0, (a.numRows - c) * a.numCols);
+            zeroRows(a, a.numRows - c, a.numRows);
+        } else {
+            a.zero();
+        }
+    }
+
+    /**
+     * Shifts all rows down by the specified number of places
```

Comment:
Could you mention what happens to the rows that are now unoccupied?

Reply:
Done.